usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/DateTime.java

```diff
@@ -0,0 +1,216 @@
+    private int seconds;
+
+    public DateTime(){}
+
+    public DateTime(String dateTime) {
+        this.year = Integer.parseInt(dateTime.substring(0, LEAP_YEAR_FREQ));
+        this.month = Integer.parseInt(dateTime.substring(MONTH_START_POS, JULY));
+        this.day = Integer.parseInt(dateTime.substring(DAY_START_POS, FIRST_TWO_DIGITS));
+        this.hour = Integer.parseInt(dateTime.substring(HOUR_START_POS, HOUR_END_POS));
+        this.minutes = Integer.parseInt(dateTime.substring(MIN_START_POS, MIN_END_POS));
+        this.seconds = Integer.parseInt(dateTime.substring(SEC_START_POS));
+    }
+
+    public DateTime(int year, int month, int day, int hour, int minutes, int seconds) throws InvalidFormatException {
+        this.year = year;
+        this.month = month;
+        this.day = day;
+        this.hour = hour;
+        this.minutes = minutes;
+        this.seconds = seconds;
+        if (isNegative(year, month, hour, minutes, seconds) || isOutOfRange(month, hour, minutes, seconds)) {
+            throw new InvalidFormatException();
+        }
+    }
+
+
+    /**
+     * Method responsible for checking whether the date values are out of range or not.
+     * @param month The month value that we want to check
+     * @param hour The hour value that we want to check
+     * @param minutes The minutes value that we want to check
+     * @param seconds The seconds value that we want to check
+     * @return True if any value is out of range or false otherwise
+     */
+    private boolean isOutOfRange(int month, int hour, int minutes, int seconds) {
+        int nDays = numberOfDays(year, month);
+        boolean dateOutOfRange = month > DECEMBER || day > nDays;
+        boolean hourOutOfRange = hour >= MAX_HOUR || minutes >= MAX_MINUTES_SECONDS || seconds > MAX_MINUTES_SECONDS;
+        return dateOutOfRange || hourOutOfRange;
+    }
+
+    /**
+     * Method responsible for checking whether the date values are negative or not.
+     * @param year The year value that we want to check
+     * @param month The month value that we want to check
+     * @param hour The hour value that we want to check
+     * @param minutes The minutes value that we want to check
+     * @param seconds The seconds value that we want to check
+     * @return True if any value is negative or false otherwise
+     */
+    private boolean isNegative(int year, int month, int hour, int minutes, int seconds) {
+        boolean isYearNegative = year < 0 || month < 0 || day < 0;
+        boolean isDateNegative = hour < 0 || minutes < 0 || seconds < 0;
+        return isYearNegative || isDateNegative;
+    }
+
+    /**
+     * Method responsible for obtaining the number of days for a given month of a given year.
+     * @param year The year value
+     * @param month The month for which we want to obtain the number of days
+     * @return The number of days of the month for that given year
+     */
+    private int numberOfDays(int year, int month) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [NCSS for this method is 16 (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5314325)